### PR TITLE
ISPN-9472 FunctionalMapTest is not executed by suite

### PR DESCRIPTION
**Pull request by danberindei**
https://issues.jboss.org/browse/ISPN-9472

* Catch exceptions in ChainedMethodInterceptor and replace them with fake test failures
* Fix parameters in FunctionalStorageTypeTest

### CODE REVIEW

File: core/src/test/java/org/infinispan/test/AbstractInfinispanTest.java [hunk 1]
```diff
@@ -98,8 +100,12 @@
                String instanceName = ((AbstractInfinispanTest) instance).getTestName();
                Object otherInstance = instancesByName.putIfAbsent(instanceName, instance);
                if (otherInstance != null) {
-                  System.err.printf("ERROR: Duplicate test name: %s, classes %s and %s", instanceName,
-                        instance.getClass().getName(), otherInstance.getClass().getName());
+                  String message = String.format("Duplicate test name: %s, classes %s and %s", instanceName,
+                                                 instance.getClass().getName(), instance.getClass().getName());
```
Comment (anistor) [inline core/src/test/java/org/infinispan/test/AbstractInfinispanTest.java]: The last param should be ```otherInstance.getClass().getName()```

File: tasks/scripting/src/main/java/org/infinispan/scripting/impl/ScriptMetadata.java [hunk 2]
```diff
@@ -134,7 +134,7 @@ public String toString() {
       Optional<String> combiner = Optional.empty();
       Optional<String> collator = Optional.empty();
       Optional<String> reducer = Optional.empty();
-      MediaType dataType = MediaType.APPLICATION_OBJECT;
+      MediaType dataType = MediaType.APPLICATION_UNKNOWN;
```
Comment (danberindei) [inline tasks/scripting/src/main/java/org/infinispan/scripting/impl/ScriptMetadata.java]: I'm not sure if this is the right approach @tristantarrant @gustavonalle, but it does fix `ScriptingDataStoresTest`

**Comment by tristantarrant** [comment]
Hm, 29 failures.

**Comment by danberindei** [comment]
@tristantarrant maybe replacing `MediaType.APPLICATION_OBJECT` with `MediaType.APPLICATION_UNKNOWN` wasn't a good idea after all... any suggestions? /cc @gustavonalle

**Comment by danberindei** [comment]
Updated to remove the script media type change, I see you already have https://issues.jboss.org/browse/ISPN-9474 for that @tristantarrant

**Comment by anistor** [comment]
The main culprit was solved by Gustavo. Please rebase this @danberindei and let's see what a new CI run indicates.

**Comment by tristantarrant** [comment]
Merged, thanks

**Pull request merged**